snark/src/r1cs.rs

```diff
@@ -1,70 +1,48 @@
-use ark_relations::r1cs::{R1CS, ConstraintSynthesizer};
+use crate::SNARK;
+use ark_ff::Field;
+use ark_relations::r1cs::{ConstraintMatrices, ConstraintSynthesizer, Instance, Witness, R1CS};
+use ark_std::rand::{CryptoRng, RngCore};
 
-/// The basic functionality for a SNARK.
+/// A [`SNARKForR1CS`] is a [`SNARK`] for the [`R1CS`] relation.
 pub trait SNARKForR1CS<F: Field>: SNARK<R1CS<F>> {
-    /// Generates a proof of satisfaction of the arithmetic circuit C (specified
-    /// as R1CS constraints).
-    fn prove<C: ConstraintSynthesizer<F>, Rng: RngCore + CryptoRng>(
-        circuit_pk: &Self::ProvingKey,
-        witness_generator: C,
+    /// Generate inputs for the SNARK indexer from [`cs`].
```

Comment:
```suggestion
    /// Generate the matrices for the SNARK indexer from [`cs`].
```

Reply:
Or perhaps we should keep this line, and a second line in the comments for each of these, specifying what the inputs are?

Reply:
I like that latter idea.

Reply:
Done.